docs/guides/tfhers.md

```diff
@@ -0,0 +1,340 @@
+circuit.keygen()
+# since both inputs have the same type, they will use the same secret key, thus we serialize it once
+secret_key: bytes = tfhers_bridge.serialize_input_secret_key(input_idx=0)
+# we write it to a file to be used by TFHE-rs
+with open("secret_key_from_concrete", "wb") as f:
+    f.write(secret_key)
+```
+
+Next, we generate client and server keys in TFHE-rs using the shared secret key from Concrete. We will cover serialization in a [later section](#serialization-of-ciphertexts-and-keys), so there's no need to worry about how we loaded the secret key.
+
+```rust
+use tfhe::core_crypto::prelude::LweSecretKey;
+use tfhe::ClientKey;
+
+/// ...
+
+let lwe_sk: LweSecretKey<Vec<u64>> = load_lwe_sk("secret_key_from_concrete");
+let shortint_key =
+    tfhe::shortint::ClientKey::try_from_lwe_encryption_key(
+        lwe_sk,
+        // Concrete uses this parameters to define the TFHE-rs ciphertext type
+        tfhe::shortint::prelude::PARAM_MESSAGE_2_CARRY_3_KS_PBS
+    ).unwrap();
+let client_key = ClientKey::from_raw_parts(shortint_key.into(), None, None);
+let server_key = client_key.generate_server_key();
+```
+
+#### KeyGen starts in TFHE-rs (use case 1.2)
+
+First, we generate the TFHE-rs keyset and then serialize the shared secret key that will be used to encrypt the inputs
+
+```rust
+use tfhe::{prelude::*, ConfigBuilder};
+use tfhe::generate_keys;
+
+/// ...
+
+// Concrete uses this parameters to define the TFHE-rs ciphertext type
+let config = ConfigBuilder::with_custom_parameters(
+        tfhe::shortint::prelude::PARAM_MESSAGE_2_CARRY_3_KS_PBS,
+    )
+.build();
+
+let (client_key, server_key) = generate_keys(config);
+let (integer_ck, _, _) = client_key.clone().into_raw_parts();
+let shortint_ck = integer_ck.into_raw_parts();
+let (glwe_secret_key, _, _) = shortint_ck.into_raw_parts();
+let lwe_secret_key = glwe_secret_key.into_lwe_secret_key();
+
+save_lwe_sk(lwe_secret_key, "secret_key_from_tfhers");
+```
+
+Next, we generate a Concrete keyset using the shared secret key from TFHE-rs.
+
+```python
+# this was generated from TFHE-rs
+with open("secret_key_from_tfhers", "rb") as f:
+    sk_buff = f.read()
+# maps input indices to their secret key
+input_idx_to_key = {0: sk_buff, 1: sk_buff}
+# we do a Concrete keygen starting with an initial set of secret keys
+tfhers_bridge.keygen_with_initial_keys(input_idx_to_key_buffer=input_idx_to_key)
+```
+
+### Using Ciphertexts
+
+We should have everything now to encrypt, compute and decrypt on both sides. Whether you started keygen in Concrete (use case 1.1) or in TFHE-rs (use case 1.2) doesn't matter. Both cases will endup with compatible keysets on both sides. We will now do an encryption and computation in TFHE-rs, move to Concrete, run our circuit, and come back to TFHE-rs for decryption.
+
+First, we do encryption and a simple addition in TFHE-rs. For more information on how to save ciphertexts, refer to [Serialization](#serialization-of-ciphertexts-and-keys).
```

Comment:
"First, we do encryption and a simple addition in TFHE-rs. " : I am a bit lost, are we still with the (x+y) % 213 example? Why do we define the function several times? Isn't the function done in Concrete (ie not in TFHE-rs)?